tyr-runner/src/test/java/org/jboss/tyr/ci/TeamCityCITest.java

```diff
@@ -15,12 +15,55 @@
  */
 package org.jboss.tyr.ci;
 
+import org.jboss.tyr.TestUtils;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 
 public class TeamCityCITest {
 
-    @Test(expected = IllegalArgumentException.class)
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Test
     public void testTCInitWithoutPropertiesSet() {
+        exception.expect(IllegalArgumentException.class);
         new TeamCityCI().init();
     }
+
+    @Test
+    public void testTriggerBuildNullParameters() {
+        exception.expect(NullPointerException.class);
+        new TeamCityCI().triggerBuild(null);
+    }
+
+    @Test
+    public void testTriggerFailedTestsNullParameters() {
+        exception.expect(NullPointerException.class);
+        new TeamCityCI().triggerFailedTests(null);
+    }
+
+    @Test
+    public void testTriggerBuildPrPayloadMissingPullRequestParameter() {
+        exception.expect(NullPointerException.class);
+        new TeamCityCI().triggerBuild(TestUtils.EMPTY_PAYLOAD);
```

Comment:
shouldn't this raise something like "Invalid payload exception"? Also for other tests after this one.